inst/templates/ad_adpet.R

```diff
@@ -0,0 +1,189 @@
+# Name: ADPET
+#
+# Label: PET Scan Analysis Dataset
+#
+# Input: adsl, nv,
+
+library(admiral)
+library(pharmaversesdtm) # Contains example datasets from the CDISC pilot project
+library(dplyr)
+library(lubridate)
+library(stringr)
+
+# Load source datasets ----
+
+# Use e.g. `haven::read_sas()` to read in .sas7bdat, or other suitable functions
+# as needed and assign to the variables below.
+# For illustration purposes read in admiral test data
+nv <- admiralneuro::nv_neuro
+adsl <- admiral::admiral_adsl
+
+nv <- convert_blanks_to_na(nv)
+
+# Lookup tables ----
+
+# Assign PARAMCD, PARAM, and PARAMN
+param_lookup <- tibble::tribble(
+  ~NVTESTCD, ~NVCAT, ~NVLOC, ~PARAMCD, ~PARAM, ~PARAMN,
+  "SUVR", "FBP", "NEOCORTICAL COMPOSITE", "SUVRFBP", "FBP Standard Uptake Ratio Neocortical Composite", 1,
+  "SUVR", "FBB", "NEOCORTICAL COMPOSITE", "SUVRFBB", "FBB Standard Uptake Ratio Neocortical Composite", 2,
+  "SUVR", "FTP", "NEOCORTICAL COMPOSITE", "SUVRFTP", "FTP Standard Uptake Ratio Neocortical Composite", 3,
+  "VR", "FBP", NA, "VRFBP", "FBP Qualitative Visual Classification", 4,
+  "VR", "FTP", NA, "VRFTP", "FTP Qualitative Visual Classifcation", 5
+)
+attr(param_lookup$NVTESTCD, "label") <- "NV Test Short Name"
+
+# Derivations ----
+
+# Get list of ADSL vars required for derivations
+adsl_vars <- exprs(TRTSDT, TRTEDT, TRT01A, TRT01P)
+
+adpet <- nv %>%
+  # Join ADSL with NV (need TRTSDT for ADY derivation)
+  derive_vars_merged(
+    dataset_add = adsl,
+    new_vars = adsl_vars,
+    by_vars = exprs(STUDYID, USUBJID)
+  ) %>%
+  ## Calculate ADT, ADY ----
+  derive_vars_dt(
+    new_vars_prefix = "A",
+    dtc = NVDTC
+  ) %>%
+  derive_vars_dy(reference_date = TRTSDT, source_vars = exprs(ADT))
+
+adpet <- adpet %>%
+  ## Add PARAMCD only - add PARAM etc later ----
+  derive_vars_merged_lookup(
+    dataset_add = param_lookup,
+    new_vars = exprs(PARAMCD, PARAM),
+    by_vars = exprs(NVTESTCD, NVCAT, NVLOC)
+  ) %>%
+  ## Calculate AVAL and AVALC ----
+  # AVALC should only be mapped if it contains non-redundant information.
+  mutate(
+    AVAL = NVSTRESN,
+    AVALC = ifelse(
+      is.na(NVSTRESN) | as.character(NVSTRESN) != NVSTRESC,
+      NVSTRESC,
+      NA
+    )
+  )
+
+## Get visit info ----
+# See also the "Visit and Period Variables" vignette
+# (https://pharmaverse.github.io/admiral/articles/visits_periods.html#visits)
+adpet <- adpet %>%
+  # No Timing variables so we can remove?
+  mutate(
+    # ATPTN = NVTPTNUM,
+    # ATPT = NVTPT,
+    AVISIT = case_when(
+      str_detect(VISIT, "SCREEN|UNSCHED|RETRIEVAL|AMBUL") ~ NA_character_,
+      !is.na(VISIT) ~ str_to_title(VISIT),
+      TRUE ~ NA_character_
+    ),
+    AVISITN = as.numeric(case_when(
+      VISIT == "BASELINE" ~ "0",
+      str_detect(VISIT, "WEEK") ~ str_trim(str_replace(VISIT, "WEEK", "")),
+      TRUE ~ NA_character_
+    )),
+    BASETYPE = "LAST"
+  )
+
+## Calculate ONTRTFL ----
+adpet <- derive_var_ontrtfl(
+  adpet,
```

Comment:
I'd rather pipe `adpet` into `derive_var_ontrtfl()` as it's easier for a user to move this block elsewhere if need be.